doc/report.md

```diff
@@ -143,133 +213,332 @@ Di seguito sono riportati i requisiti individuati durante lo studio del dominio
+inserimento delle mutazioni e dei fattori, oltre che per avere una controprova visiva del corretto funzionamento delle
+classi di model e dell'engine.
 
-Per quanto riguarda invece il package `it.unibo.pps.bunny.view` non sono stati implementati test di `ScalaTest` per verificarne il funzionamento, in quanto si è deciso di effettuare prevalentemente del beta-testing, cioè di eseguire direttamente l'applicativo per verificare ad esempio il movimento dei coniglietti e la gestione dei pannelli per l'inserimento delle mutazioni e dei fattori, oltre che per avere una controprova visiva del corretto funzionamento delle classi di model e dell'engine.
+#### Spadoni
+
+### Rocco
 
-## Retrospettiva
+Nello sviluppo del progetto inizialmente mi sono occupata della realizzazione grafica della board di gioco in maniera
+parallela a Baiardi attraverso, prima lo studio e in seguito l'utilizzo della
+libreria [ScalaFX](https://www.scalafx.org/) e lo strumento di
+supporto [SceneBuilder](https://www.oracle.com/java/technologies/javase/javafxscenebuilder-info.html) necessario per la
+traduzione di quanto strutturato graficamente in codice `fxml`. In particolare gli elementi di GUI che ho interamente
+realizzato sono il pannello dei fattori disturbanti e l'orologio, insieme alla logica del suo funzionamento legata alla
+simulazione.
+
+Per quanto riguarda l'identificazione di una parte di model interamente a me riconducibile è la parte legata alle
+mutazioni, che hanno consentito poi di mettere mano sia alla creazione dei loro relativi test che alle parti di
+controller legate al loro funzionamento all'interno della simulazione.
+
+Le parti implementate comprendono:
+
+* `it.unibo.pss.bunny.model.bunny.Mutation`
+* `it.unibo.pss.bunny.model.world.Reproduction` (solo la logica d'introduzione delle mutazioni durante la riproduzione)
+* il package `it.unibo.pss.bunny.view.scalaFX.components` (ad esclusione del file `BunnyView` e il sotto
+  package `charts`)
+* `it.unibo.pss.bunny.view.scalaFX.FXControllers.FactorsPanelController`
+* `it.unibo.pss.bunny.view.scalaFX.FXControllers.BaseAppController` (in collaborazione con tutti gli altri componenti
+  del gruppo)
+
+#### TDD
+
+Per l'implementazione del model delle mutazioni mi sono rifatta alla tecnica **TDD** cercando di seguire il ciclo
+**Red-Green-Refactor**, in modo tale da produrre prima un test fallimentare (**Red**) che consentisse di determinare
+_cosa_ realizzare, di far passare poi i test (**Green**) e infine mettendo mano allo stile del codice in un secondo
+momento (**Refactor**). Questa tecnica ha permesso di testare il funzionamento corretto delle mutazioni senza
+considerare il modo, quindi il _come_, in cui di fatto le mutazioni sia state implementate. Il codice per questo è
+estendibile e ha consentito nel corso dello sviluppo di mantenere traccia della correttezza del funzionamento globale di
+tutte le feature e nella preventiva intercettazione di eventuali bug.
```

Comment:
In questa sezione hai scritto praticamente le stesse cose scritte dall'asia, non ha senso scriverlo due volte